### PR TITLE
Feature algorithm

### PR DESCRIPTION
**Pull request by SevlaMare**
-Initial Setup
-Make algorithm work
-Create both methods
-Improve performance
-Linter fixes

-Linter configuration update
Since Rubocop 0.81.0 update, some changes are required
More details on https://docs.rubocop.org/en/latest/versioning/

### CODE REVIEW

**Review by mosesogwo — CHANGES_REQUESTED** [review]
**Hi @SevlaMare,:clap: good job on this project but there is a problem with your bubble_sort method :fast_forward:**

Your bubble_sort method throws an error.
`p bubble_sort([4, 3, 78]) => wrong number of arguments (given 0, expected 2..3) (ArgumentError)`

_**Kindly address the error and submit another code review request.**_

**Any question? You can reach me on Slack (Moses Ogwo)**

**Cheers:stars:**

### Status: Changes Requested.

**Comment by SevlaMare** [comment]
I got no error this test case on VScode:
error

Neither on reply, in the live version:
https://repl.it/@ThiagoMiranda2/bubblesortV0

I think was because I paste the wrong link on review request.

File: bubble_sort.rb [hunk 1]
```diff
@@ -1,28 +1,45 @@
...
+  end
+  array
 end
 
-# second optimization
-# the most right element will be sorted each pass
-# next pass only get unsorted elements, loop over until (n-1)
+p bubble_sort_by(%w[hi hello hey]) { |left, right| left.length - right.length }
+p bubble_sort_by(%w[hi hello hey by]) { |left, right| left.length - right.length }
+p bubble_sort_by(%w[b a aaa aa aaa a aa]) { |left, right| left.length - right.length }
```
Comment (stratospherique) [inline bubble_sort.rb]: - All commented/debugging code should be deleted. :eyes:

**Review by stratospherique — APPROVED** [review]
### Great work @SevlaMare :clap: The methods are working. Thus, the project is approved :heavy_check_mark: You may merge :+1: :+1: 

Keep up the good work :muscle: 
**_Ahmed Mahfoudh_**

**Pull request merged**